### PR TITLE
Refactor and developer enhancements added

### PR DESCRIPTION
**Pull request by greg-szabo**

### CODE REVIEW

File: QUICKSTART.md [hunk 1]
```diff
@@ -21,5 +21,5 @@ Reminders for the time when you already know what you're doing.
 
 ### Quick start a Nightking AMI build
 - Double-check the `experiments` folder to see if everything is there.
-- Run `packer build packer.json` with AWS access.
+- Run `NOAUTORUN=1 packer build packer.json` with AWS access.
```
Comment (adizere) [inline QUICKSTART.md:24]: My usual workflow is to start an instance, not a build. Was wondering if the `NOAUTORUN` option is also available upon starting a Nightking instance scenario -- would be great if so! I guess this option can be passed, just as `DEBUG`, as a tag.

File: roles/terraform-templates/files/terraform-templates-modules-0a1/server/01-input.tf [hunk 2]
```diff
@@ -59,6 +54,17 @@ variable experiments {
   type = "string"
 }
 
+variable dev {
```
Comment (adizere) [inline roles/terraform-templates/files/terraform-templates-modules-0a1/server/01-input.tf:57]: Ah, I guess this is what enables `NOAUTORUN` (regarding my comment above)?

File: roles/scripts/files/nightking.bash [hunk 3]
```diff
@@ -27,58 +37,73 @@ do
...
+      log terraform_build 1
+      export TERRAFORM_RESULT=0
+      terraform apply --auto-approve -no-color || export TERRAFORM_RESULT=$?
+    # ... reinitialize a previous infra.
     else
-      if [ -d /etc/experiments/"${XP}"/terraform ]; then
-        stemplate /etc/experiments/"${XP}"/terraform -o /root/terraform-"${XP}" --env -f /etc/experiments/"${XP}"/config.toml --all
-      fi
+      IP_LIST="$(terraform show /root/terraform-${PREVIOUS_INFRA}/terraform.tfstate -no-color | grep "^  public_ip =" | sed "s/^  public_ip = //" | tr '\n' ' ')"
+      sudo -u ec2-user pssh -p 20 -t 10 -i -O "StrictHostKeyChecking no" -H "${IP_LIST}" sudo nohup /usr/local/sbin/run-role.bash "${XP}" &
```
Comment (thanethomson) [inline roles/scripts/files/nightking.bash]: Nice! 🚀  So just to clarify, this enclosing if/else statement checks whether the network's already been deployed by Terraform: if not, it deploys it (which automatically runs the `run-role.bash` script); if it has already been deployed, it uses `pssh` to execute `run-role.bash` on the target machines?

**Review by thanethomson — DISMISSED** [review]
Haven't tried running it yet, but it looks good to me! 🍻

File: QUICKSTART.md (hunk 1 above)
Comment (greg-szabo) [reply QUICKSTART.md:24]: DEV=1 will do that as a tag. NOAUTORUN is more for end-users with special needs. (Might be removed later.)

File: roles/terraform-templates/files/terraform-templates-modules-0a1/server/01-input.tf (hunk 2 above)
Comment (greg-szabo) [reply roles/terraform-templates/files/terraform-templates-modules-0a1/server/01-input.tf:57]: yup.

File: roles/scripts/files/nightking.bash (hunk 3 above)
Comment (greg-szabo) [reply roles/scripts/files/nightking.bash]: Yes. This was recently clarified more in the script by pulling some of the code into functions. `run-bash` was renamed to `runxp` for easier typing.

**Pull request merged**